src/resolvers/api/getActivity.js

```diff
@@ -0,0 +1,64 @@
+const format = require('pg-format')
+const _get = require('lodash.get')
+
+const { query } = require('../utils')
+
+const getActivity = async (_, args) => {
+  const urbitId = _get(args, 'input.urbitId') || null
+  const since = _get(args, 'input.since') || null
+  const until = _get(args, 'input.until') || null
+
+  let queryString
+
+  queryString = format('select * from %I', 'ping')
+  if (since || until || urbitId) {
+    queryString += ' where'
+  }
+  if (since) {
+    queryString += format(' %I < \'%s\'', 'response_time', since)
+  }
+
+  if (until) {
+    if (since) {
+      queryString += ' and'
+    }
+    queryString += format(' %I > \'%s\'', 'response_time', until)
+  }
+
+  if (urbitId) {
+    if (since || until) {
+      queryString += ' and'
+    }
+    queryString += format(' %s=\'%s\'', 'node_id', urbitId)
+  }
+
+  queryString += format(' order by %s desc', 'response_time')
+  queryString += ';'
+
+  const getActivityResponse = await query(queryString)
+
+  const returnArr = []
+  const responseDates = []
+
+  const rows = _get(getActivityResponse, 'rows') || []
+
+  if (rows.length > 0) {
+    for (const i in rows) {
```

Comment:
Happy to change this because that probably is a better way to handle the default value assignment instead of using `_get`. However, `for...in` probably isn't a problem here itself, since the order of elements in the array being iterated over is not important. Right?

Reply:
I have now changed this for all other occurrences of the `for...in` pattern, except for one case in which I decided `for (let i = 0; i < something.length; i++)` made more sense.